### PR TITLE
(Very) Basic Debug

### PR DESCRIPTION
**Pull request by Arthi-chaud**
## Proposed changes

A single variable in the RendereSystem lets one display the current FPS

### Add all necessary labels to understand with ease the scope of this pull request

Describe here the big picture of your changes here to communicate to the others members of the group and why we should accept this pull request.

## Information
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Non-breaking changes

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,8 @@ include_directories(bomberman sources)
 add_subdirectory(${PROJECT_SOURCE_DIR}/lib/wal)
 add_subdirectory(${PROJECT_SOURCE_DIR}/lib/Ray)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
```
Comment (Octopus773) [inline CMakeLists.txt:13]: Is this really needed ?

File: sources/System/Renderer/RenderSystem.hpp [hunk 2]
```diff
@@ -21,6 +21,13 @@ namespace BBM
 
 		//! @brief The camera used to render.
 		RAY::Camera::Camera3D _camera;
+
+		//! @brief Defines if the debug informations must be displayed or not
+		bool _debugMode;
+
+		//! @brief Window framerate limit
+		static constexpr short FPS = 30;
```
Comment (Octopus773) [inline sources/System/Renderer/RenderSystem.hpp:29]: didn't we say it was 60 ?

File: sources/System/Renderer/RenderSystem.cpp [hunk 3]
```diff
@@ -20,8 +20,11 @@ namespace BBM
 		}),
 		_wal(wal),
 		_window(window),
-		_camera(Vector3f(), Vector3f(), Vector3f(0, 1, 0), 50, CAMERA_PERSPECTIVE)
-	{}
+		_camera(Vector3f(), Vector3f(), Vector3f(0, 1, 0), 50, CAMERA_PERSPECTIVE),
+		_debugMode(true)
```
Comment (Octopus773) [inline sources/System/Renderer/RenderSystem.cpp:24]: you should take it as param

File: CMakeLists.txt (hunk 1 above)
Comment (p8nut) [reply CMakeLists.txt:13]: this should be add at the end when the game is already optimised (might add some more FPS)

File: sources/System/Renderer/RenderSystem.hpp (hunk 2 above)
Comment (p8nut) [reply sources/System/Renderer/RenderSystem.hpp:29]: why does the debug set the framerate ?
your debug should only display the calculated framerate.

File: sources/System/Renderer/RenderSystem.cpp (hunk 3 above)
Comment (p8nut) [reply sources/System/Renderer/RenderSystem.cpp:24]: debug could be a friend class which then let it access to private date in your engine

File: sources/System/Renderer/RenderScreenSystem.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			typeid(CameraComponent),
+			typeid(PositionComponent)
+		}),
+		_window(window),
+		_camera(Vector3f(), Vector3f(), Vector3f(0, 1, 0), 50, CAMERA_PERSPECTIVE)
+	{
+		this->_window.setFPS(RenderScreenSystem::FPS);
+	}
+
+	void RenderScreenSystem::onSelfUpdate()
```
Comment (AnonymusRaccoon) [inline sources/System/Renderer/RenderScreenSystem.cpp:22]: This is the old renderer, it should be deleted.

**Pull request merged**